XO/XO.py

```diff
@@ -0,0 +1,123 @@
+
+import random
+
+N = 3
+EMPTY = "E"
+PLAYERS_SIGNS = ["X", "O"]
+
+
+def get_names():
+    """
+    :return: {"X": input1, "O": input2}
+    :rtype: dict
+    """
+    texts = ["Insert the name of player 0\n", "Insert the name of player 1\n"]
+    names = get_user_inputs(texts)
+    return {PLAYERS_SIGNS[0]: names[0], PLAYERS_SIGNS[1]: names[1]}
+
+
+def get_action(current_player, board):
+    """
+    :type current_player: str
+    :return: list[0]=line number, list[1]=column number
+    :rtype: list of ints
+    """
+    texts = [current_player + ", insert line number (0/1/2)\n", current_player + ", insert column number (0/1/2)\n"]
+    inputs = get_user_inputs(texts)
+    if not is_legal_position(inputs, board):
+        print("the input is not legal!")
+        inputs = get_action(current_player, board)
+    return [int(i) for i in inputs]
```

Comment:
Since the board will always be 2-dimensional, I think a tuple would be nicer here.